vimeo/VimeoNetworking
# MP-40: Add video transcode status

#### Ticket

[MP-40](https://vimean.atlassian.net/browse/MP-40)

#### Pull Request Checklist


#### Issue Summary

- Adds a video transcode status model and request builder 

#### Implementation Summary

This PR uses the new Networking API compatible with codable models
- `TranscodeStatusEndpointType`: defines the request to `/videoURI/status` 
- `VideoTranscodeStatus`: decodable type defining the video status object 
  - [API docs for Transcode status](https://developer.vimeo.com/api/reference/videos#get_transcode_status) - note, must be logged in as a Staff Account to see docs 
  - Basis for `TranscodeState` enum values: [link to backend code](https://github.vimeows.com/Vimeo/vimeo/blob/df07e4b924275827fc68e92c733d2ae795667b02/src/Vimeo/Api/Response/ClipStatusResponse.php#L51)
- `VideoTranscodeStatusTests`: tests decoder since we defined it ourselves in order to map to an enum type

#### Reviewer Tips

- Because this is a self contained request, it seemed like a safe place to use out new networking API 

- How would this code be used by a client? Pass your custom endpoint type into this method to make a request:
```swift
public func request<Model: Decodable>(
        _ endpoint: EndpointType,
        startImmediately: Bool = true,
        completionQueue: DispatchQueue = .main,
        then callback: @escaping (Result<Model, Error>) -> Void
    ) -> RequestToken 
``` 

For example 

```swift
let endpointType = TranscodeStatusEndpointType(videoURI: "videos/xxxxxx")

_ = VimeoClient.shared.request(endpointType) { (result: Result<VideoTranscodeStatus, Error>) in
   switch result {
      case .success(let result):
        // handle result
      case .failure(let error):
        // handle error
   }
}
```

#### How to Test

- Confirm added tests pass
- To test fetching data from the API 
  - find a video you own and get its uri
  - log into the account owning the video
  - run the example code provided above with the uri of the video you own
  - you should see that you get a valid `VideoTranscodeStatus` object back, such as `VideoTranscodeStatus(state: VimeoNetworking.VideoTranscodeStatus.TranscodeState.ready, progress: 100, timeLeft: 0)` for a video that is already transcoded and ready watch
